vtm-web/src/org/oscim/gdx/client/MapUrl.java

```diff
@@ -83,36 +86,39 @@ else if (p.startsWith("tilt="))
                 MercatorProjection.latitudeToY(lat),
                 1 << zoom,
                 rotation,
-                tilt);
+                tilt, roll);
 
     }
 
     @Override
     public void run() {
         mMap.viewport().getMapPosition(pos);
-        int lat = (int) (MercatorProjection.toLatitude(pos.y) * 1000);
-        int lon = (int) (MercatorProjection.toLongitude(pos.x) * 1000);
-        int rot = (int) (pos.bearing);
-        rot = (int) (pos.bearing) % 360;
-        //rot = rot < 0 ? -rot : rot;
+        float lat = (float) (MercatorProjection.toLatitude(pos.y) * 1000000);
+        float lon = (float) (MercatorProjection.toLongitude(pos.x) * 1000000);
+        float rot = ((pos.bearing) % 360) * 1000000;
+        float roll = ((pos.roll) % 360) * 1000000;
+        float tilt = pos.tilt * 1000000;
+        float zoom = (float) pos.zoomLevel * 1000000;
 
-        if (curZoom != pos.zoomLevel || curLat != lat || curLon != lon
-                || curTilt != rot || curRot != (int) (pos.bearing)) {
+        if (curZoom != zoom || curLat != lat || curLon != lon
+                || curTilt != rot || curRot != rot || curRoll != roll) {
```

Comment:
BTW why the change to floats since compare is again with fixed precision?

Reply:
Now I understood the reason why the values are compared using integer ... it was to fix the precision. I modified these just because I thought integer degrees was very low precision for bearing, tilt and rolls. But that's not the main issue. I'll roll it back.